dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/indicator/IndicatorServiceTest.java

```diff
@@ -281,4 +281,22 @@ void testNumeratorTranslation() {
     assertEquals(numeratorTranslated, indicatorA.getDisplayNumeratorDescription());
     assertEquals(denominatorTranslated, indicatorA.getDisplayDenominatorDescription());
   }
+
+  @Test
+  void testRemoveIndicator() {
```

Comment:
It doesn't look like this actually tests the new code added, whether an `Indicator` has been removed from a `Section` or not. I pulled the branch code, commented out the new addition to the `SectionDeletionHandler` and this test still passes.
Should there not be a `Section` with `Indicator`s which should then be checked after the `Indicator` is deleted?

Reply:
I can add it. It just becomes a bit too arbitrary. There is a very high number of associations between data set and other entities, and all other types of entities for that matter, and we have almost no test coverage for deletion of associations. So we could add this particular case, but it will still give a very low confidence overall, so just wondering if it is worth it.

Reply:
I don't think it's arbitrary at all. A good test confirms expected behaviour & confirms the presence of a bug if there's a regression.
If this change is in fact a fix (Indicator deletion prohibited when associated with Section) then surely we'd want to make sure that this scenario can't happen again?
This topic arises very frequently in our codebase whereby a simple test would have prevented a regression in behaviour.

Reply:
Okay no problem. Will update the test.

Moving the work to a new PR due to the signed commit requirement:

https://github.com/dhis2/dhis2-core/pull/19015